src/vs/editor/contrib/hover/browser/markdownHoverParticipant.ts

```diff
@@ -44,23 +55,56 @@ export class MarkdownHover implements IHoverPart {
 	}
 }
 
+export class VerboseMarkdownHover extends MarkdownHover {
+
+	constructor(
+		owner: IEditorHoverParticipant<MarkdownHover>,
+		range: Range,
+		contents: IMarkdownString[],
+		isBeforeContent: boolean,
+		ordinal: number,
+		public readonly provider: HoverProvider | undefined,
+		public readonly verbosityMetadata: HoverVerbosityMetadata | undefined,
+	) {
+		super(owner, range, contents, isBeforeContent, ordinal);
+	}
+}
+
+interface MarkdownFocusMetadata {
+	index: number | undefined;
+	element: HTMLElement | undefined;
+	focusRemains: boolean;
+}
+
 export class MarkdownHoverParticipant implements IEditorHoverParticipant<MarkdownHover> {
 
 	public readonly hoverOrdinal: number = 3;
+	private _anchor: HoverAnchor | undefined;
+	private _context: IEditorHoverRenderContext | undefined;
+
+	private _providers: (HoverProvider | undefined)[] = [];
+	private _focusMetadata: MarkdownFocusMetadata = {
+		index: undefined,
+		element: undefined,
+		focusRemains: false
+	};
+	private _verbosityLevels: Map<number, number> = new Map();
 
 	constructor(
 		protected readonly _editor: ICodeEditor,
 		@ILanguageService private readonly _languageService: ILanguageService,
 		@IOpenerService private readonly _openerService: IOpenerService,
 		@IConfigurationService private readonly _configurationService: IConfigurationService,
 		@ILanguageFeaturesService protected readonly _languageFeaturesService: ILanguageFeaturesService,
+		@IKeybindingService private readonly _keybindingService: IKeybindingService
 	) { }
 
 	public createLoadingMessage(anchor: HoverAnchor): MarkdownHover | null {
-		return new MarkdownHover(this, anchor.range, [new MarkdownString().appendText(nls.localize('modesContentHover.loading', "Loading..."))], false, 2000);
+		return new VerboseMarkdownHover(this, anchor.range, [new MarkdownString().appendText(nls.localize('modesContentHover.loading', "Loading..."))], false, 2000, undefined, undefined);
 	}
 
 	public computeSync(anchor: HoverAnchor, lineDecorations: IModelDecoration[]): MarkdownHover[] {
+		this._anchor = anchor;
```

Comment:
Yes so I save the `anchor` in order to extract the position where to request the hover with additional information. I changed the code a bit to save only the position since that is what we ultlimately want. I wonder if there is a better way to rewrite this, so that we don't also have to save the position